File: LeetCode/1441.py
#Link: https://leetcode.com/problems/build-an-array-with-stack-operations/

class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        operations = []
        i, j = 1, 0
        while(i <= n and j < len(target)):
            operations.append("Push")
            if i != target[j]:
                operations.append("Pop")
                j -= 1
            i += 1
            j += 1
        return operations

File: LeetCode/cpp/1441.cpp
//Link: https://leetcode.com/problems/build-an-array-with-stack-operations/

class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> operations;
        for(int i = 1, j = 0 ; i <= n && j < target.size() ; i++, j++) {
            operations.push_back("Push");
            if(i != target[j]) {
                operations.push_back("Pop");
                j--;
            }
        }
        return operations;
    }
};
